include/util.h

```diff
@@ -85,11 +85,11 @@
 
 #define BITS_PER_LONG 32
 
-#define GENMASK(h, l) ({ 										\
-		uint32_t t = (~0UL);									\
-		t = t << (BITS_PER_LONG - (h - l + 1));					\
-		t = t >> (BITS_PER_LONG - (h + 1));						\
-		t;														\
+#define GENMASK(h, l) ({ 					\
+		uint32_t t = (uint32_t)(~0UL);			\
```

Comment:
I'am not sure, but I think you can just remove remove the L to avoid the warning.

Reply:
I think some platforms could define U as a 16-bit and UL as 32-bit in which case this would only work up to 16-bits. Making it UL and then explicitly casting to 32-bit is safest in my opinion.

Reply:
true